tests/realtikvtest/addindextest1/cross_ks_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package addindextest
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/pingcap/tidb/pkg/config/kerneltype"
+	"github.com/pingcap/tidb/pkg/disttask/framework/handle"
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/meta/model"
+	"github.com/pingcap/tidb/pkg/testkit"
+	"github.com/pingcap/tidb/pkg/testkit/testfailpoint"
+	"github.com/pingcap/tidb/tests/realtikvtest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddIndexOnSystemTable(t *testing.T) {
+	if kerneltype.IsClassic() {
+		t.Skip("This test is only for nextgen kernel, skip it in classic kernel")
+	}
+
+	t.Run("bootstrap system keyspace", func(t *testing.T) {
```

Comment:
can we merge all those part? they seems belongs to a single test, `t.Run` is for running sub-test

Reply:
`t.Run` is used to simulate multiple instances at process level. There are problems if we merge it to a single test.